File: frontend/src/api/schedule-task.ts
import http from './http'

const request = {
  get: <T = any>(url: string, config?: any) => http.get<T>(url, config),
  post: <T = any>(url: string, data?: any, config?: any) => http.post<T>(url, data, config),
  put: <T = any>(url: string, data?: any, config?: any) => http.put<T>(url, data, config),
  delete: <T = any>(url: string, config?: any) => http.delete<T>(url, config)
}

export interface ScheduleTask {
  id: number
  name: string
  active: number
  cron: string
  workflow: string
  exec_token: string
  created_at: string
  updated_at: string
  deleted_at?: string
  cron_entry_id: number
  category: string
  dcn: string
  next_run_time?: string
  cron_desc: string
  status_text: string
}

export interface ScheduleTaskRequest {
  name: string
  active: number
  cron: string
  workflow: string
  exec_token: string
  category: string
  dcn: string
}

export interface ScheduleTaskQuery {
  name?: string
  active?: number
  category?: string
  dcn?: string
  start_time?: string
  end_time?: string
  page?: number
  page_size?: number
  order_by?: string
  order_dir?: string
}

export interface BatchOperationRequest {
  ids: number[]
  operation: 'enable' | 'disable' | 'delete'
}

export interface ScheduleTaskListResponse {
  list: ScheduleTask[]
  total: number
  page: number
  page_size: number
}

// 获取任务列表
export const getScheduleTaskList = (params: ScheduleTaskQuery) => {
  return request.get<ScheduleTaskListResponse>('/system/schedule-tasks', { params })
}

// 获取任务详情
export const getScheduleTaskById = (id: number) => {
  return request.get<ScheduleTask>(`/system/schedule-tasks/${id}`)
}

// 创建任务
export const createScheduleTask = (data: ScheduleTaskRequest) => {
  return request.post<ScheduleTask>('/system/schedule-tasks', data)
}

// 更新任务
export const updateScheduleTask = (id: number, data: ScheduleTaskRequest) => {
  return request.put<ScheduleTask>(`/system/schedule-tasks/${id}`, data)
}

// 删除任务
export const deleteScheduleTask = (id: number) => {
  return request.delete(`/system/schedule-tasks/${id}`)
}

// 更新任务状态
export const updateScheduleTaskStatus = (id: number, active: number) => {
  return request.put(`/system/schedule-tasks/${id}/status`, { active })
}

// 立即执行任务
export const executeScheduleTask = (id: number) => {
  return request.post(`/system/schedule-tasks/${id}/execute`)
}

// 批量操作
export const batchOperateScheduleTasks = (data: BatchOperationRequest) => {
  return request.post('/system/schedule-tasks/batch', data)
}

// 获取分类列表
export const getScheduleTaskCategories = () => {
  return request.get<string[]>('/system/schedule-tasks/categories')
}

// 获取数据中心列表
export const getScheduleTaskDCNs = () => {
  return request.get<string[]>('/system/schedule-tasks/dcns')
}

File: backend/internal/model/schedule_task.go
package model

import (
	"time"
)

// ScheduleTask 调度任务模型 (基于wecom_tasks表)
type ScheduleTask struct {
	ID          uint       `gorm:"primaryKey;autoIncrement" json:"id"`
	Name        string     `gorm:"not null;size:100;comment:任务名称" json:"name"`
	Active      int        `gorm:"default:0;comment:是否可用" json:"active"`
	Cron        string     `gorm:"not null;size:100;comment:定时任务表达式" json:"cron"`
	Workflow    string     `gorm:"not null;size:100;comment:工作流ID" json:"workflow"`
	ExecToken   string     `gorm:"not null;size:100;comment:执行token" json:"exec_token"`
	CreatedAt   *time.Time `gorm:"comment:创建时间" json:"created_at"`
	UpdatedAt   *time.Time `gorm:"comment:更新时间" json:"updated_at"`
	DeletedAt   *time.Time `gorm:"comment:删除时间" json:"deleted_at"`
	CronEntryID int64      `gorm:"not null;default:0" json:"cron_entry_id"`
	Category    string     `gorm:"not null;size:32;comment:任务分类" json:"category"`
	DCN         string     `gorm:"not null;size:100;comment:数据中心" json:"dcn"`
}

// TableName 指定表名
func (ScheduleTask) TableName() string {
	return "wecom_tasks"
}

// ScheduleTaskRequest 创建/更新任务请求
type ScheduleTaskRequest struct {
	Name      string `json:"name" binding:"required,max=100"`
	Active    int    `json:"active"`
	Cron      string `json:"cron" binding:"required,max=100"`
	Workflow  string `json:"workflow" binding:"required,max=100"`
	ExecToken string `json:"exec_token" binding:"required,max=100"`
	Category  string `json:"category" binding:"required,max=32"`
	DCN       string `json:"dcn" binding:"required,max=100"`
}

// ScheduleTaskQuery 查询参数
type ScheduleTaskQuery struct {
	Name       string `form:"name"`
	Active     *int   `form:"active"`
	Category   string `form:"category"`
	DCN        string `form:"dcn"`
	StartTime  string `form:"start_time"`
	EndTime    string `form:"end_time"`
	Page       int    `form:"page,default=1"`
	PageSize   int    `form:"page_size,default=10"`
	OrderBy    string `form:"order_by,default=id"`
	OrderDir   string `form:"order_dir,default=desc"`
}

// ScheduleTaskResponse 任务响应
type ScheduleTaskResponse struct {
	*ScheduleTask
	NextRunTime   *string `json:"next_run_time"`
	CronDesc      string  `json:"cron_desc"`
	StatusText    string  `json:"status_text"`
}

// BatchOperationRequest 批量操作请求
type BatchOperationRequest struct {
	IDs       []uint `json:"ids" binding:"required"`
	Operation string `json:"operation" binding:"required,oneof=enable disable delete"`
}